src/test/database/test_level2db.py

```diff
@@ -134,3 +155,58 @@ def test_count_failed_scans_with_limit(self, level2db):
     @pytest.mark.slow
     def test_count_failed_scans_with_offset(self, level2db):
         assert level2db.count_failed_scans(FREQMODE, offset=1) == 1
+
+
+class TestGetMeasurements:
+    @pytest.mark.parametrize("min_scanid,limit,expect", (
+        (7014791071, 50, 36),
+        (7014791071, 5, 5),
+        (7014791072, 20, 0),
+    ))
+    def test_get_measurements_respect_offset_and_limit(
+            self, level2db_with_example_data, min_scanid, limit, expect):
+        products = [
+            "ClO / 501 GHz / 20 to 50 km",
+            "O3 / 501 GHz / 20 to 50 km",
+        ]
+        measurements = level2db_with_example_data.get_measurements(
+            products, min_altitude=None, max_altitude=None,
+            min_pressure=None, max_pressure=None,
+            start_time=None, end_time=None, areas=None,
+            fields=None, min_scanid=min_scanid, document_limit=limit
+        )
+        results = list(measurements)
+        assert len(results) == expect
+
+    @pytest.mark.parametrize("min_scanid,limit,expect", (
+        (7014791071, 37, 2),
+        (7014791071, 36, 0),
```

Comment:
this is wrong I guess then, the expect must be 36. However you are missing the case of limit being 35...

Reply:
this is correct. expect is the number of collapsed products, limit is on database documents. in this case 36 db documents are collapsed into two products with same scanid. if the limit is reached I do not include data from the last scanid. I have introduced a min_limit of 1000 on the server, so it will never be a problem that data from a single scanid hit the limit.

Reply:
missing test of the returned next_min_scanid

Reply:
added